### ISSUE TITLE
Per key RGB FX

### ISSUE THREAD

**Issue by DragonWoot** [WootingKb/wootility-issues#178]
**Is your feature request related to a problem? Please describe.**
I'd like some keys animated with certain RGB FX and some with others, I think another issue talked about layering RGB FX, like wave at 50% brightness that brightens surrounding keys a la jelly/aoe when a key is pressed, also some keys just jelly/aoe and some wave or other effect on other parts of the board. Probably the best way is to separate out the keyboard animations from the key press interactive FX and allow per key layering of both

**Describe the solution you'd like**
See above

**Describe alternatives or workarounds you've considered**

**Additional context**

### Additional information

**Platform information**
e.g. Windows 10/MacOS Big Sur/Linux

**Wootility and Firmware version**
Find in your Wootility Settings

**Keyboard Type**
Wooting 2HE and Wooting 60HE ARM

**Comment by bverkron** [comment]
There are actually some productivity reasons for this kind of feature. For example on my Rocatt Vulcan TKL Pro I would have certain keys light up blue _and_ use the "breathe" effect to indicate they have to be pressed in sequence for certain actions like Ctrl+K, Ctrl+] meaning you press CTRL+K, let go of K then press ]. Having just shortcut keys that were lit up solid blue wouldn't give enough info as there are too many others lit up the same color that do not need to be pressed in sequence. With per-key effects you can sort of get multi-layer RGB if that make sense. Blue color for keys invovled in shortcuts and blue + effect for ones that need to be pressed in sequence.

You could argue that you could just use another color for keys that need to be pressed in sequence but that can lead to a more visually cluttered and confusing color scheme and sometimes other colors are used for other things.

**Comment by BigBrainAFK** [comment]
Im not sure if full multilayer RGB effects are possible (as in multiple different RGB effects per key on 1 layer). Usually I would recommend using Artemis2 for this as they dedicate all work to pure RGB related functionality.

**Comment by BigBrainAFK** [comment]
Yes Artemis is windows only until the avalonia release is finished when it would be fully Cross-platform. The main issue with the multiple rgb layers in 1 function layer is that it would require more ram and more processing. Im unsure how feasible it is to implement but im sure Simon can pitch in on that
